code/script.js

```diff
@@ -246,98 +249,170 @@ const generateBoard = () => {
 const setSecret = () => {
-  secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+  let randomElement = Math.random() * charactersInPlay.length //note to self! for a randomised number rounded off to length of array (will be a decimal (float) and therefore math.floor after)
+  let randomElementInt = Math.floor(randomElement); 
+  secret = charactersInPlay[randomElementInt]; //note to self! here we assign secret person by picking out the randomised number (aka person) from the array
+  //secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+  //note to self! straight brackets show that we are picking out a randomised element on the randomised position in our array
 }
 
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  charactersInPlay = CHARACTERS;
+  generateBoard();
+  // What else should happen when we start the game? 
+  setSecret();
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  const category = questions.options[questions.selectedIndex].parentNode.label;
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
+  const property = questions.options[questions.selectedIndex].label;
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
+      value: property,
       // 👆 add the value from the input here
       category: category,
     }
   } else if (category === 'eye color') {
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: property,
+      // 👆 add the value from the input here
+      category: category,
     // Set this up your self
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
+      attribute: property,
       // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
       value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
       category: category,
     }
   } else if (category === 'other') {
+    currentQuestion = {
+      attribute: property,
+      value: true,
+      // 👆 add the value from the input here
+      category: category,
     // Set this up your self (should be same structure as above)
+    }
   }
 }
 
 // This function should be invoked when you click on 'Find Out'.
 const checkQuestion = () => {
   // Compare the currentQuestion with the secret person.
   // See if we should keep or remove people based on that
+  let keepPerson;
+  let checking = currentQuestion.attribute;
```

Comment:
It's the same in this line, the value of checking is not being reassigned. In this case, is `const`